diff --git a/toolchains/cc/cc.nix b/toolchains/cc/cc.nix
index e340d51..c636b3b 100644
--- a/toolchains/cc/cc.nix
+++ b/toolchains/cc/cc.nix
@@ -135,7 +135,7 @@ pkgs.runCommand "bazel-${cc.orignalName or cc.name}-toolchain"
     # Determine toolchain tool paths.
     #
     # If a tool is not available then we use `bin/false` as a stand-in.
-    declare -A TOOLS=( [ar]=ar [cpp]=cpp [dwp]=dwp [gcc]=cc [gcov]=gcov [llvm-cov]=llvm-cov [ld]=ld [libtool]=libtool [nm]=nm [objcopy]=objcopy [objdump]=objdump [strip]=strip )
+    declare -A TOOLS=( [ar]=ar [cpp]=cpp [dwp]=dwp [gcc]=cc [gcov]=gcov [llvm-cov]=llvm-cov [ld]=ld [libtool]=libtool [nm]=nm [objcopy]=objcopy [objdump]=objdump [strip]=strip [cpp-module-deps-scanner]=cc-scan-deps-wrapper )
     TOOL_NAMES=(''${!TOOLS[@]})
     declare -A TOOL_PATHS=()
     for tool_name in ''${!TOOLS[@]}; do
