diff --git a/core/private/cc_toolchain/unix_cc_toolchain_config.bzl b/core/private/cc_toolchain/unix_cc_toolchain_config.bzl
index e500999..d4ce0bd 100644
--- a/core/private/cc_toolchain/unix_cc_toolchain_config.bzl
+++ b/core/private/cc_toolchain/unix_cc_toolchain_config.bzl
@@ -1864,6 +1864,8 @@ def _impl(ctx):
             user_link_flags_feature,
             default_link_libs_feature,
             external_include_paths_feature,
+            includes_feature,
+            include_paths_feature,
             fdo_optimize_feature,
             dbg_feature,
             opt_feature,
