diff --git a/toolchains/cc/cc.bzl b/toolchains/cc/cc.bzl
index 5c28f41..fecea2a 100644
--- a/toolchains/cc/cc.bzl
+++ b/toolchains/cc/cc.bzl
@@ -323,7 +323,8 @@ def nixpkgs_cc_configure(
         cc_lang = "c++",
         cc_std = "c++0x",
         cross_cpu = "",
-        apple_sdk_path = ""):
+        apple_sdk_path = "",
+        cc_stdlib = None):
     """Use a CC toolchain from Nixpkgs. No-op if not a nix-based platform.
 
     By default, Bazel auto-configures a CC toolchain from commands (e.g.
@@ -397,6 +398,7 @@ def nixpkgs_cc_configure(
       cc_std: string, `"c++0x"` by default. Used to populate `CXX_FLAG` so the compiler uses the given language standard. Can be set to `"none"` together with appropriate `copts` in the `cc_library` call: see above.
       cross_cpu: string, `""` by default. Used if you want to add a cross compilation C/C++ toolchain. Set this to the CPU architecture for the target CPU. For example x86_64, would be k8.
       apple_sdk_path: string, `""` by default. Obtain the default nix `apple-sdk` for the toolchain form the Nix expression under this attribute path.  Uses default repository if no `nix_file` or `nix_file_content` is provided.
+      cc_stdlib: string, optional. Override the default C++ standard library.
     """
     nixopts = list(nixopts)
     nix_file_deps = list(nix_file_deps)
@@ -458,6 +460,9 @@ def nixpkgs_cc_configure(
             "ccStd",
             cc_std,
         ])
+
+    if cc_stdlib:
+        nixopts.extend(["--argstr", "stdlib", cc_stdlib])
     
     if apple_sdk_path:
         nixopts.extend(["--argstr", "appleSDKPath", apple_sdk_path])
diff --git a/toolchains/cc/cc.nix b/toolchains/cc/cc.nix
index 5504fbe..5b4df46 100644
--- a/toolchains/cc/cc.nix
+++ b/toolchains/cc/cc.nix
@@ -10,6 +10,7 @@
   ccLang ? "c++",
   ccStd ? "c++0x",
   appleSDKPath ? "apple-sdk",
+  stdlib ? null,
 }:
 
 let
@@ -257,11 +258,14 @@ pkgs.runCommand "bazel-${cc.orignalName or cc.name}-toolchain"
     )
     LINK_LIBS=(
       ${
-        # Use stdenv.isDarwin as a marker instead of cc.isClang because
-        # we might have usecases with stdenv with clang and libstdc++.
-        # On Darwin libstdc++ is not available, so it's safe to assume that
-        # everybody use libc++ from LLVM.
-        if stdenv.isDarwin then "-lc++" else "-lstdc++"
+        if stdlib == null then 
+          # Use stdenv.isDarwin as a marker instead of cc.isClang because
+          # we might have usecases with stdenv with clang and libstdc++.
+          # On Darwin libstdc++ is not available, so it's safe to assume that
+          # everybody use libc++ from LLVM.
+          if stdenv.isDarwin then if cc.isClang then "-lc++" else "-lstdc++" else "-lstdc++"
+        else
+          "-l${stdlib}"
       }
       -lm
     )
